test/e2e/appstudio_test.go

```diff
@@ -0,0 +1,203 @@
+			user.token = req.GetToken()
+
+			VerifyResourcesProvisionedForSignup(t, awaitilities, user.signup, "base")
+			_, err = hostAwait.GetMasterUserRecord(wait.WithMurName(user.username))
+			require.NoError(t, err)
+
+			// since the registration service always provisions users to the default tier users need to be
+			// promoted to the appstudio tier in order to test appstudio scenarios
+			t.Run("promote to appstudio tier", func(t *testing.T) {
+				// given
+				changeTierRequest := NewChangeTierRequest(hostAwait.Namespace, user.signup.Status.CompliantUsername, promotionTier)
+
+				// when
+				err = hostAwait.CreateWithCleanup(context.TODO(), changeTierRequest)
+
+				// then
+				require.NoError(t, err)
+				_, err := hostAwait.WaitForChangeTierRequest(changeTierRequest.Name, toBeComplete)
+				require.NoError(t, err)
+				VerifyResourcesProvisionedForSignup(t, awaitilities, user.signup, promotionTier)
+
+				// then - wait until ChangeTierRequest is deleted by our automatic GC
+				err = hostAwait.WaitUntilChangeTierRequestDeleted(changeTierRequest.Name)
+				assert.NoError(t, err)
+			})
+
+			t.Run("use proxy to create a configmap in the user appstudio namespace via proxy API", func(t *testing.T) {
+				// given
+				cmName := fmt.Sprintf("%s-test-cm", user.username)
+				expectedCM := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      cmName,
+						Namespace: user.username,
+					},
+					Data: map[string]string{
+						"planet": "venus",
+					},
+				}
+
+				// when
+				proxyCl := hostAwait.CreateAPIProxyClient(user.token)
+				err := proxyCl.Create(context.TODO(), expectedCM)
+				require.NoError(t, err)
+
+				// then
+				createdCM := &corev1.ConfigMap{}
+				err = proxyCl.Get(context.TODO(), types.NamespacedName{Namespace: user.username, Name: cmName}, createdCM)
+				require.NoError(t, err)
+				require.NotEmpty(t, createdCM)
+				require.Equal(t, "venus", createdCM.Data["planet"])
+			})
+
+			t.Run("try to create a resource in an unauthorized namespace", func(t *testing.T) {
+				// given
+				cmName := fmt.Sprintf("%s-appstudio-test-cm", user.username)
+				expectedCM := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      cmName,
+						Namespace: hostAwait.Namespace, // user should not be allowed to create a resource in the host operator namespace
+					},
+					Data: map[string]string{
+						"planet": "venus",
+					},
+				}
+
+				// when
+				proxyCl := hostAwait.CreateAPIProxyClient(user.token)
+
+				// then
+				err := proxyCl.Create(context.TODO(), expectedCM)
+				require.EqualError(t, err, fmt.Sprintf(`configmaps is forbidden: User "system:serviceaccount:%[1]s:appstudio-%[1]s" cannot create resource "configmaps" in API group "" in the namespace "%[2]s"`, user.username, hostAwait.Namespace))
+			})
+
+			if index == 1 { // only for the second user
+				t.Run("try to create a resource in the other users namespace", func(t *testing.T) {
```

Comment:
it doesn't matter if it's for the first one or the second one because the namespace will be deleted by the cleanup anyway.

Reply:
Hmm, good point. I didn't notice that. But now that you recommended to remove the "delete usersignup and expect all resources to be deleted" step I can verify it for the second user. I added a check to ensure the first user's namespace still exists before trying to create the configmap.

Reply:
cool :+1: